plugins/Referrers/Columns/Base.php

```diff
@@ -139,6 +147,46 @@ protected function getReferrerInformation($referrerUrl, $currentUrl, $idSite, Re
         return $referrerInformation;
     }
 
+    protected function getExcludedReferrers($idSite): array
+    {
+        try {
+            $attributes = Cache::getCacheWebsiteAttributes($idSite);
+
+            if (isset($attributes['excluded_referrers'])) {
+                return $attributes['excluded_referrers'];
+            }
+        } catch (UnexpectedWebsiteFoundException $e) {
+            $cached = Cache::getCacheGeneral();
+            if (isset($cached['global_excluded_referrers'])) {
+                return $cached['global_excluded_referrers'];
+            }
+        }
+
+        return [];
+    }
+
+    protected function isReferrerExcluded(string $referrerUrl): bool
+    {
+        $urlsByHost = [];
+        $parsedReferrer = @parse_url($referrerUrl);
+
+        $excludedUrls = new SiteUrls();
+        $excludedReferrers = $this->getExcludedReferrers($this->idsite);
+        foreach ($excludedReferrers as $excludedReferrer) {
+            // ensure referrer is prefixed with protocol, so a parse_url works
+            $excludedReferrer = 'https://' . preg_replace('/^https?:\/\//', '', $excludedReferrer);
```

Comment:
Could this result in `https://http://example.org` if the excluded referrer is using `http` ?

Reply:
Same as in javascript. It removes `http://` and `https://`